npc/re/jobs/2e/rebellion.txt

```diff
@@ -0,0 +1,1082 @@
+	} else if (.@select == 6) {
+		.@questComplete = 12344;
+	}
+	for (.@i = 12340; .@i <= 12345; .@i++) {
+		erasequest(.@i);
+		if (.@i <= .@questComplete) {
+			setquest(.@i);
+			completequest(.@i);
+		}
+	}
+	close;
+
+OnInit:
+	disablenpc(strnpcinfo(NPC_NAME));
+	end;
+}
+
+job_gun,56,26,0	script	rebelroom#1	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("einbroch", 49, 97);
+	end;
+}
+
+job_gun,18,10,0	script	rebelroom#2	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("job_gun", 81, 29);
+	end;
+}
+
+job_gun,165,22,0	script	rebelroom#3	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("job_gun", 196, 35);
+	end;
+}
+
+job_gun,197,49,0	script	rebelroom#4	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("job_gun", 157, 21);
+	end;
+}
+
+job_gun,88,144,0	script	shootingrange#1	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 93, 123);
+	end;
+}
+
+job_gun,105,97,0	script	shootingrange#2	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 125, 98);
+	end;
+}
+
+job_gun,156,115,0	script	shootingrange#3	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 157, 137);
+	end;
+}
+
+job_gun,152,154,0	script	shootingrange#4	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 135, 154);
+	end;
+}
+
+job_gun,130,146,0	script	shootingrange#5	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 120, 130);
+	end;
+}
+
+job_gun,165,18,1	script	Furnace Controller#1	CLEAR_NPC,{
+	if (Class != Job_Gunslinger) {
+		mes("Only authorized user can do the operation.");
+		close;
+	}
+	if (getstatus(SC_MONSTER_TRANSFORM, 1) != G_L_SHECIL) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	if (questprogress(12342) != 1) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	mes("Do you want to dredge the mold up from the furnace?");
+	next;
+	if (select("Quit.", "Begin.") == 1) {
+		mes("Switch to the Standby mode.");
+		close;
+	}
+	hideonnpc("Furnace Controller#1");
+	donpcevent("Furnace Controller#2::OnStart");
+	mes("Initializing Furnace Controller ZX-3100...");
+	close;
+
+OnInit:
+	hideoffnpc("Furnace Controller#1");
+	end;
+}
+
+job_gun,214,36,1	script	Auto Anvil#1	CLEAR_NPC,{
+	if (Class != Job_Gunslinger) {
+		mes("Only authorized user can do the operation.");
+		close;
+	}
+	if (countitem(Tiny_Flame) > 0) {
+		mes("Unauthorized firearm creation has been detected. Report to Manager Elwin.");
+		close;
+	}
+	if (getstatus(SC_MONSTER_TRANSFORM, 1) != G_L_SHECIL || countitem(Steel_Article_) <= 4 || questprogress(12342) != 1) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	
+	mes("Do you want to use the auto anvil and shape the standard Steel Artifacts into a firearm?");
+	next;
+	if (select("Quit.", "Begin.") == 1) {
+		mes("For your safety, please step away from the auto anvil.");
+		close;
+	}
+	if (rand(1, 10) > 6) {
+		specialeffect(EF_REPAIRWEAPON);
+		mes("You have successfully created a firearm using the auto anvil. ^ff0000Please report before your transformation expires^000000.");
+		delitem(Steel_Article_, 5);
+		getitem(Tiny_Flame, 1);
```

Comment:
please a weight check for `Tiny_Flame` at the top of this npc

Reply:
done